lang/funcs/collection_test.go

```diff
@@ -2336,6 +2336,20 @@ func TestSlice(t *testing.T) {
 			listWithUnknowns,
 			cty.NumberIntVal(1),
 			cty.NumberIntVal(2),
+			cty.ListVal([]cty.Value{cty.UnknownVal(cty.String)}),
+			false,
+		},
+		{ // unknowns in the list
+			listWithUnknowns,
+			cty.NumberIntVal(0),
+			cty.NumberIntVal(2),
+			listWithUnknowns,
+			false,
+		},
+		{ // unknowns list
```

Comment:
Similar question here about the comment being more useful to show what these cases are testing